test/unit/lib/meli-client.test.js

```diff
@@ -11,7 +11,8 @@ const Account = require('../../../model/account')
 // Mount db connection
 require('../../../config/db').connect()
```

Comment:
The mongo database connection should be a self-managed singleton of the APIs you don't connect manually to the database when using, for example, Spring Data.

service.query() -> if not connected, connected, save singleton connection and query.
				-> if connected, query.

Reply:
I get your point. Mongoose Model already handles what you say by default. But... in this case the idea is to eagerly initialize the connection to notice possible connection errors early (since in the app there are many services that require initialization, not only this one). Otherwise the whole app might start up but then will fail on the first query due to a bad/faulty db connection...

Reply:
I can see what you say, what if you code a `db.ping()` or still use the `connect()` approach but prepare the CRUD functions to connect in case that isn't connected already, so you don't need to `connect()` everywhere.

Reply:
I got the first part but not quite the second... What do you mean by CRUD functions? Mongoose ODM already provides Model access out of the box + requests cache. So if the DB is down the requests still happen but are enqueued until the DB is up again. Also it will automatically attempt to connect if it's the first time and it was not connected, or if it was connected and lost connection. The idea of explicitly calling `.connect()` is eagerly detecting possible errors or inconsistencies in the connection (ie. listening to 'onError' or 'onConnect' events), but it's not required for the rest of the app to work. At most, it's only required to be done once at the beginning of the app

Reply:
What happens if the DB is working at the beginning of the process and at the middle it goes down? then you aren't handling error connections? Using `connect()` to have a free-do-any-query isn't handling all the posible errors, you should be aware of that on any time of the code, so the `.connect()` may only work to check if it's up to avoid setting up the server but you have to still check every time you query anything. Although you don't need to "check" if database if working on a controlled environment (testing should be always a controlled environment, that means that it has to be as less as outside apis or db dependent as we can.